Utils/src/ParticleMapper.cc

```diff
@@ -1,62 +1,52 @@
 #include "MitPhysics/Utils/interface/ParticleMapper.h"
 
+#include "TVector2.h"
+
 ClassImp(mithep::ParticleMapper)
 
 using namespace mithep;
 
 //--------------------------------------------------------------------------------------------------
-ParticleMapper::ParticleMapper() :
-  fDeltaEta(0),
-  fDeltaPhi(0),
-  fNumParticles(0),
-  fNumEtaBins(0),
-  fNumPhiBins(0),
-  fNumTotBins(0),
-  fParticleLocation(0),
-  fBinContents(0)
-{ }
+ParticleMapper::ParticleMapper(Double_t DeltaEta/* = 0.3*/, Double_t DeltaPhi/* = 0.3*/, Double_t EtaMax/* = 5.*/) :
+  fDeltaEta(DeltaEta),
+  fDeltaPhi(DeltaPhi),
+  fEtaMax(EtaMax),
+  fNumEtaBins(2*ceil(EtaMax/fDeltaEta)),
+  fNumPhiBins(floor(2*(TMath::Pi()/fDeltaPhi))),  // The last bin will be larger than fDeltaPhi
+  fParticleLocation(),
+  fBinContents(fNumEtaBins * fNumPhiBins)
+{
+}
 
 //--------------------------------------------------------------------------------------------------
 ParticleMapper::~ParticleMapper()
 {
-  delete fParticleLocation;
-  delete fBinContents;
```

Comment:
These lines are removed because both variables are now vectors. Nevertheless, even if they remained arrays, you would have to do
delete [] fParticleLocation;
delete [] fBinContents;
. Array new must be followed by array delete, otherwise weird memory errors will occur.